File: front/src/models/item.ts
import Color from "./color";
import Country from "./country";

class Item {
    id?: number;
    name: string;
    quantity: number;
    countryId: number;
    colorId: number;
    active: boolean;
    country?: Country;
    color?: Color;

    constructor(name: string, quantity: number, countryId: number, colorId: number, active: boolean, id?: number, country?: Country, color?: Color) {
        this.name = name;
        this.quantity = quantity;
        this.countryId = countryId;
        this.colorId = colorId;
        this.active = active;
        this.id = id;
        this.color = color;
        this.country = country;
    }
}

export default Item;

File: back/src/main/java/com/hahn/challenge/domain/entities/Item.java
package com.hahn.challenge.domain.entities;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column
    private String name;
    @Column
    private Integer quantity;
    @Column
    private Boolean active;
    @Column
    private Integer countryId;
    @Column
    private Integer colorId;
    @ManyToOne
    @JoinColumn(name = "countryId", insertable = false, updatable = false)
    private Country country;
    @ManyToOne
    @JoinColumn(name = "colorId", insertable = false, updatable = false)
    private Color color;
}
